tour_model_eval/trips_in_bins_and_clusters.ipynb

```diff
@@ -42,293 +49,789 @@
+    "\n",
+    "user = all_users[1]"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "hollow-prefix",
+   "id": "finnish-istanbul",
    "metadata": {},
    "outputs": [],
    "source": [
-    "logging.debug('There are %s users, we will plot the graphs for one of them' % len(all_users))\n",
-    "\n",
-    "user = all_users[1]"
+    "# we should experiment with different values here\n",
+    "radius = 300"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "prospective-conducting",
+   "id": "dependent-machinery",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "trips = pipeline.read_data(uuid=user)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "steady-consultation",
+   "metadata": {},
+   "source": [
+    "Note: Run trips on all data and trips in bins/clusters above cutoff point separately "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "exotic-timing",
    "metadata": {},
+   "source": [
+    "## Visualizing trips from bins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "recovered-peoples",
+   "metadata": {},
+   "source": [
+    "### Trips from all bins"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ultimate-bottom",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "sim = similarity.similarity(trips, radius)\n",
+    "sim.bin_data()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "amino-particular",
+   "metadata": {
+    "scrolled": true
+   },
    "outputs": [],
    "source": [
-    "import emission.analysis.modelling.tour_model.cluster_pipeline as eamtc"
+    "logging.debug('There are %s bins before filtering' % len(sim.bins))"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "finnish-istanbul",
+   "id": "reasonable-sharp",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "#Create colormap for all bins\n",
+    "cmp_all_bins = cm.linear.Set1_07.to_step(len(sim.bins),index=[i for i in range (len(sim.bins))])\n",
+    "cmp_all_bins"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "incorporated-plymouth",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# we should experiment with different values here\n",
-    "radius = 300"
+    "# Plot all the bin trips on the map, use different color for different bins.\n",
+    "m = folium.Map(location=[trips[0].data.start_loc[\"coordinates\"][1], trips[0].data.start_loc[\"coordinates\"][0]],zoom_start=12, max_zoom= 30, control_scale=True)\n",
+    "for t in range (len(sim.bins)):    \n",
+    "    for i in range (len(sim.bins[t])):\n",
+    "        layer = folium.PolyLine([[trips[sim.bins[t][i]].data.start_loc[\"coordinates\"][1],trips[sim.bins[t][i]].data.start_loc[\"coordinates\"][0]],\n",
```

Comment:
can you add some comments here on how you are selecting trips to help future students? I would recommend pulling out the current trip selection into a variable so you don't repeat yourself. You should also use `for bin in bins` instead of the index unless you actually need the index, so 

```
for t, curr_bin in sim.bins:
   for curr_trip_index in curr_bin:
      curr_trip = trips[curr_trip_index]
....
```

Much easier to understand and less convoluted